deployer/helm_upgrade_decision.py

```diff
@@ -0,0 +1,358 @@
+    associated support chart values files have been modified. To be parsed to GitHub
+    Actions in order to generate parallel jobs in a matrix.
+    Note: "cluster folders" are those that contain a cluster.yaml file.
+    Args:
+        modified_dirpaths (list[path obj]): List of absolute paths to cluster folders
+            that contain EITHER a modified cluster.yaml OR a modified support.values.yaml
+            file
+        upgrade_all_clusters (bool, optional): If True, generates jobs to update the
+            support chart on all clusters. This is triggered when common config has been
+            modified in the support helm chart. Defaults to False.
+    Returns:
+        list[dict]: A list of dictionaries. Each dictionary contains: the name of a
+            cluster and the cloud provider that cluster runs on.
+    """
+    # Empty list to store the matrix definitions in
+    matrix_jobs = []
+
+    if upgrade_all_clusters:
+        print(
+            "Common config has been updated. Generating jobs to upgrade support chart on ALL clusters."
+        )
+
+        # Overwrite cluster_filepaths to contain paths to all clusters
+        if test_env:
+            # We are running a test via pytest. We only want to focus on the cluster
+            # folders nested under the `tests/` folder.
+            modified_dirpaths = [
+                filepath.parent
+                for filepath in Path(os.getcwd()).glob("**/cluster.yaml")
+                if "tests/" in str(filepath)
+            ]
+        else:
+            # We are NOT running a test via pytest. We want to explicitly ignore the
+            # cluster folders nested under the `tests/` folder.
+            modified_dirpaths = [
+                filepath.parent
+                for filepath in Path(os.getcwd()).glob("**/cluster.yaml")
+                if "tests/" not in str(filepath)
+            ]
+
+    for cluster_filepath in modified_dirpaths:
+        # Read in the cluster.yaml file
+        with open(cluster_filepath.joinpath("cluster.yaml")) as f:
+            cluster_config = yaml.load(f)
+
+        # Generate a dictionary-style job entry for this cluster
+        cluster_info = {
+            "cluster_name": cluster_config.get("name", {}),
+            "provider": cluster_config.get("provider", {}),
+        }
+
+        # Double-check that support is defined for this cluster. If it is, append the
+        # job definition to the list of matrix jobs.
+        support_config = cluster_config.get("support", {})
+        if support_config:
+            matrix_jobs.append(cluster_info)
+        else:
+            print(f"No support defined for cluster: {cluster_config.get('name', {})}")
+
+    return matrix_jobs
+
+
+def update_github_env(hub_matrix_jobs, support_matrix_jobs):
+    """Update the GITHUB_ENV environment with two new variables describing the matrix
+    jobs that need to be run in order to update the support charts and hubs that have
+    been modified.
```

Comment:
So the decision is made to print rich tables if EITHER the `--pretty-print` argument is passed OR `GITHUB_ENV` is found in the environment:

https://github.com/2i2c-org/infrastructure/pull/1093/files#diff-a3ba5c5035ddef3b5c64742f652b5c0e14075ed68b82ee882c884d514b259817R267-R268

> I'm also thinking perhaps we should fail loudly and early if we end up running functions that assume we are doing it as part of a GitHub workflow execution - but aren't.

So we don't fail, but instead print out human-readable output instead of trying to write to the GITHUB_ENV file